extension-androidauto/src/main/java/com/mapbox/maps/extension/androidauto/MapboxCarMapSurface.kt

```diff
@@ -0,0 +1,35 @@
+package com.mapbox.maps.extension.androidauto
+
+import androidx.car.app.CarContext
+import androidx.car.app.SurfaceContainer
+import com.mapbox.maps.MapSurface
+import com.mapbox.maps.Style
+
+/**
+ * This contains the Android Auto head unit map information.
+ * @see MapboxCarMap.registerObserver
+ *
+ * @property carContext reference to the context provided to the [MapboxCarMap]
+ * @property mapSurface Mapbox controllable interface
+ * @property surfaceContainer A container for the Surface created by the car.
+ * @property style reference to the Mapbox Style
+ */
+class MapboxCarMapSurface internal constructor(
+  val carContext: CarContext,
+  val mapSurface: MapSurface,
+  val surfaceContainer: SurfaceContainer,
+  val style: Style
```

Comment:
Do we actually need to keep a reference to the Style here? It's accessible anyway via `surface.getMapboxMap().getStyle()` and this reference is guaranteed to be the style that is actually used in the map.

Reply:
True but that is also nullable. When it is included as a non-nullable object inside this object, your downstream implementations appear to be deterministic

Feels better
``` kotlin
        mapboxCarMapSurface.style.addPersistentStyleCustomLayer(
            LogoWidget.LOGO_WIDGET_LAYER_ID,
            logoWidget.host,
            layerPosition
        )
```

Alternative 
``` kotlin
        mapboxCarMapSurface.mapSurface.getMapboxMap().getStyle()?.addPersistentStyleCustomLayer(
            LogoWidget.LOGO_WIDGET_LAYER_ID,
            logoWidget.host,
            layerPosition
        )
```

Reply:
What bothers me is that the style seem to be not updated if it's changed directly via MapboxMap derectly. Thus we'll have multiple source of truth for the current style, with hard to catch and debug bugs after that.

BTW instead of using nullable style `mapboxMap.getStyle { }` can be used that guarantees returning valid loaded style.

Reply:
I think style API is part of the mapboxMap and shouldn't be wrapped further in the maps SDK android-auto extension. We recommend users to use the async `mapboxMap.getStyle { }` to access the style.

Reply:
This sounds reasonable to say, there shouldn't be another wrapper. I'll remove as we iterate on other solutions.

The issue that this has been addressing, is observing when style changes between dark/light. There are many components that need to re-load. I see the `getStyle`, but is there an `observeStyle` somewhere?

Reply:
Seems `mapboxMap.addOnStyleDataLoadedListener` should do the trick

Reply:
Resolving this discussion and moving here https://github.com/mapbox/1tap-android/issues/1594